e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js

```diff
@@ -504,106 +500,179 @@ describe("issues 20845, 25031", () => {
-    cy.log("Test dashboard");
-    cy.get("@dashboardId").then(dashboardId => {
-      visitDashboard(dashboardId);
+    it("should allow a dashboard filter to map to a boolean field filter parameter in static embedding (metabase#27643)", () => {
+      cy.log("Test the dashboard");
+      cy.get("@dashboardId").then(dashboardId => {
+        visitDashboard(dashboardId);
+      });
+
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
+
+      cy.log("Test the embedded dashboard");
+      cy.get("@dashboardId").then(dashboard => {
+        visitEmbeddedPage({
+          resource: { dashboard },
+          params: {},
+        });
+      });
+
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
+
+      cy.log("Test the public dashboard");
+      cy.signOut();
+      cy.get("@dashboardPublicUuid").then(uuid => {
+        cy.visit(`/public/dashboard/${uuid}`);
+      });
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
     });
+  });
 
-    toggleFilterWidgetValues(["false"]);
-    getDashboardCard().findByText("Rows 1-6 of 455").should("be.visible");
+  describe("questions", () => {
+    beforeEach(() => {
+      cy.get("@postgresInvoicesExpectedInvoiceId").then(fieldId => {
+        createNativeQuestion(getQuestionDetails(fieldId), {
+          wrapId: true,
+          idAlias: "questionId",
+        });
+      });
 
-    cy.log("Test embedded dashboard");
-    cy.get("@dashboardId").then(dashboardId => {
-      visitEmbeddedPage({
-        resource: { dashboard: dashboardId },
-        params: {},
+      cy.get("@questionId").then(questionId => {
+        cy.request("POST", `/api/card/${questionId}/public_link`).then(
+          ({ body: { uuid } }) => {
+            cy.wrap(uuid).as("questionPublicUuid");
+          },
+        );
       });
     });
 
-    /**
-     * If we set the filter value now, this request won't be cancelled.
-     * And it will override the result from the next request with filter value.
-     *
-     * This seems to only happen in Cypress since it click elements on the screen
-     * a lot faster than human. I tested this manually, and the requests were cancelled properly.
-     */
-    getDashboardCard()
-      .findByText("Rows 1-6 of first 2000")
-      .should("be.visible");
-
-    toggleFilterWidgetValues(["false"]);
-    getDashboardCard()
-      .findByText("There was a problem displaying this chart.")
-      .should("not.exist");
+    it("should allow a native question filter to map to a boolean field filter parameter in static embedding (metabase#27643)", () => {
+      cy.log("Test the question");
+      cy.get("@questionId").then(questionId => {
+        visitQuestion(questionId);
+      });
```

Comment:
Nit:
Both `visitQuestion` and `visitDashboard` have been updated to work directly with aliases as well.

```suggestion
      visitQuestion("@questionId");
```

Reply:
But I think this whole block is redundant. You should just `visitQuestion: true` in the `beforeEeach` block